Katello/katello
# Fixes #37050 - restrict module copying to source repository

#### What are the changes introduced in this pull request?
Ensures that module streams are only copied from the source repository during CV publish.

#### Considerations taken when implementing this change?
Had to check that other content types aren't also affected.

#### What are the testing steps for this pull request?
From the BZ:

Steps to Reproduce:
1. Create a content view containing 1 modular and 1 non-modular repository (IE: 'Red Hat Enterprise Linux 8 for x86_64 - AppStream RPMs 8' & 'Red Hat Ansible Engine 2 for RHEL 8 x86_64 RPMs')
2. Create a content-view filter for module streams with the following settings:
- Include: true
3. Add a rule to this CV filter to:
- include all module streams
- original-module-streams: false
- Apply to all repositories in the CV: true
4. Publish a new version
5. See that module streams + modular RPMs got copied to the non-modular repo
6. Patch in my fix and publish again. See that the non-modular repo stays non-modular.
